refgenconf/refgenconf.py

```diff
@@ -2087,10 +2162,134 @@ def _select_genomes(self, genome=None, strict=False, order=None,
+                        asset_v["tags"]["default"]["seek_keys"][seek_k] = seek_v.replace(
+                            k, d)
+
+                    if (asset_k == "fasta" and "asset_children" in asset_v["tags"]["default"]):
+                        for i in range(len(asset_v["tags"]["default"]["asset_children"])):
+                            asset_v["tags"]["default"]["asset_children"][i] = asset_v["tags"]["default"]["asset_children"][i].replace(
+                                k, d)
+                    elif asset_k != "fasta":
+                        for i in range(len(asset_v["tags"]["default"]["asset_parents"])):
+                            asset_v["tags"]["default"]["asset_parents"][i] = asset_v["tags"]["default"]["asset_parents"][i].replace(
+                                k, d)
+                # use the genome digest as primary keys
+                config["genomes"][d] = config["genomes"].pop(k)
+                # remove old "genome_digest" section
+                del config["genomes"][d]["genome_digest"]
+                # change the config_version
+                config["config_version"] = target_version
+                # write over the config file
+                config.write()
+            return config
+        # restructure the genome_folder
+
+        def alter_file_tree(config):
+            my_genome = {}
+            for k, v in config["genomes"].items():
+                my_genome.update([(v["aliases"][0], k)])
+            os.mkdir(config["genome_folder"]+'/data/')
+            os.mkdir(config["genome_folder"]+'/alias/')
+            # move folder
+            for root, dirs, files in os.walk(config["genome_folder"]):
+                for dir in dirs:
+                    if dir in my_genome:
+                        shutil.move(config["genome_folder"]+"/" + dir,
+                                    config["genome_folder"]+'/data/' + dir)
+                del dirs[:]
+
+            for root, dirs, files in os.walk(config["genome_folder"]+'/data/'):
+                for dir in dirs:
+                    _swap_names_in_tree(root+dir, my_genome[dir], dir)
+                    os.mkdir(config["genome_folder"]+'/alias/'+dir)
+                    # create symlink for alias folder
+                    for genome, assets, files in os.walk(root+my_genome[dir]):
+                        for asset in assets:
+                            old_path = os.path.join(genome, asset)
+                            new_path = old_path.replace(
+                                my_genome[dir], dir).replace("/data/", "/alias/")
+                            os.mkdir(new_path)
+                        for file in files:
+                            old_path = os.path.join(genome, file)
+                            new_path = old_path.replace(
+                                my_genome[dir], dir).replace("/data/", "/alias/")
+                            os.symlink(old_path, new_path)
+                del dirs[:]
```

Comment:
The data has to be moved in a more secure way, as @nsheff suggested here: https://github.com/refgenie/refgenie/issues/203#issuecomment-693418119

Reply:
86dc8b8